llvm/include/llvm/ObjectYAML/DXContainerYAML.h

```diff
@@ -74,16 +74,35 @@ struct ShaderHash {
 };
 
 #define ROOT_ELEMENT_FLAG(Num, Val) bool Val = false;
+
+struct RootConstantsYaml {
+  uint32_t Register;
+  uint32_t Space;
+  uint32_t NumOfConstants;
+};
+
+struct RootParameterYamlDesc {
+  dxbc::RootParameterType Type;
+  dxbc::ShaderVisibility Visibility;
+  uint32_t Offset;
+
+  RootParameterYamlDesc() = default;
```

Comment:
```suggestion
```
I'm not sure what the default default constructor would do here, since none of the members are initialized to anything.